dashboard/src/reducers/endpointReducer.js

```diff
@@ -0,0 +1,20 @@
+import { SET_ENDPOINTS } from "../actions/types";
+
+const initialState = {
+  endpoints: {},
+};
+
+const EndpointReducer = (state = initialState, action = {}) => {
+  const { type, payload } = action;
+  switch (type) {
+    case SET_ENDPOINTS:
+      return {
+        ...state,
+        endpoints: Object.assign({}, payload),
+      };
```

Comment:
Just out of curiosity, why did you opt to use `Object.assign()` here instead of a simple, shallow copy (which I _think_ does the same thing):
```:
      return {
        ...state,
        endpoints: {...payload},
      };
```
and, because it's my current hobby horse (😀), I have to ask:  would it work to make `endpoints` be a _reference_ to `payload` instead of a copy of it?

Reply:
If the new state is different, the reducer must create new object, and making a `copy ` is a way to describe the unchanged part.  When there are only a few properties to keep, we can specify unchanged properties instead of copying.